### ISSUE TITLE
periodic functions runner leaks memory

### ISSUE THREAD

**Issue by chriskuehl** [ocf/ocfweb#108]
![](http://i.fluffy.cc/TmM6Mx9Cnzpwp842R6mgphr45CtfxdN4.png)

currently using 4126MB of memory on coma

it gets restarted every deploy, but if we go too long without deploying, puppet can't run (no memory) so we can't deploy anymore.

**Comment by jvperrin** [comment]
Not sure what's leaking memory here, I can't seem to reproduce this in development, it just stays at a steady memory level. I tried using [mem_top](https://pypi.python.org/pypi/mem_top), but I didn't see any memory growth in development and I can't try it in production.

**Comment by chriskuehl** [comment]
I think @daradib and I fixed this in dc434831c433fb2fa329543f4e89769e2f57ed25.

We used `objgraph` to print objects and tried running each periodic function repeatedly (removing delays in the runner to speed up the effect) and seeing which had problems. The only that was leaking objects was the daily graph one, which is my fault for not knowing how to use matplotlib.

so far things look happy:

![](http://i.fluffy.cc/QsVkllxpSNhHB2vNS4QsFcNG6Rtf33F0.png)
